server/src/main/java/org/apache/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -587,6 +594,43 @@ private StringFullResponseHolder fetchLookupsForTier(String tier) throws Interru
     void handle(Map<String, LookupExtractorFactoryContainer> lookupMap) throws Exception;
   }
 
+  private static class StatusNotice implements Notice
```

Comment:
Why do we need a separate `StatusNotice`? Can the isReady() check based behavior be integrated into the LoadNotice itself?

Reply:
Hi @a2l007, thanks for looking into this. I did try to put all into the existing LoadNotice. And I tried to split the LoadNotice to make the code more readable.

Reply:
Could you please add comments for this noticetype?